### PR TITLE
Add Commands for Syncing Production and Staging

### LINKED ISSUE #1040: Create Script to Copy data from Production to Stage
At midnight dump s3 and fixture to the sync bucket created with ansible

At 6am copy from the sync bucket to staging.

Write this using pure python  so that it can run on lambda.

The database is transfered using fixtures and flushing the database.  This might cause issues with IDs getting bigger so we might need to avoid that.

### PR DESCRIPTION
**Pull request by glassresistor**
Closes #1040

### What does this do and why?

This lets staging sync from productions data. It copies the production database to a protected folder on Amazon's S3, and then loads that data into the staging database.

### Testing & Checks

What does this PR include?
- [x] new dependencies
- [ ] migrations
- [ ] data migrations
- [ ] celery tasks
- [x] changes to environment variables or local settings
- [x] configuration changes for 3rd party services (Travis, Heroku, AWS, Github, etc) 
- [ ] visual or style changes

_Are there any human checks that should be used to verify these changes, such as user acceptance tests (UAT) or manual deployment checks?_
- [ ] before deploying, make a database snapshot and mark the time (to revert s3 to) immediately before we run the commands next checks
- [ ] After deploy, check that the commands work using heroku run and then use
    ```sh
    aws s3 ls $bucket_name --recursive | sort
    ```
    to verify that prod and staging each have the same contents before and after running
- [ ] after deploy, check that pdfs can be uploaded and downloaded from the admin
- [ ] Once verified, Add the commands to heroku scheduler on production and staging respectively

_Any integration or unit tests?_

No 

### Details

This creates two new management commands in an app called `helper_commands` (top level folder). Both of these commands use [`aws-cli`](https://github.com/aws/aws-cli) which has been added as a new dependency.

- `upload_data` dumps a fixture and uploads it to s3
- `download_data` syncs the media buckets and downloads the database and
loads its

These commands depend on some new environment settings:
- `ORIGIN_MEDIA_BUCKET_FOR_SYNC`  [only on staging do not put on production]
- `AWS_STORAGE_BUCKET_NAME`
- `SYNC_BUCKET`
- `SYNC_FIXTURE_LOCATION `
- `AWS_CLI_LOCATION`
- `SYNC_AWS_ID`
- `SYNC_AWS_ID`
- `SYNC_AWS_KEY`

### CODE REVIEW

**Comment by glassresistor** [comment]
I haven't updated the production and staging settings files yet because I'm waiting on ticket #1117  to close.  Then i will be able to test this works IRL using my local machine.

**Comment by bengolder** [comment]
If there is any breakage of the media bucket, we would likely see messages about missing pdfs in the logs, which would be triggered by one of two actions:
- opening an application to SF
- applying to SF

**Comment by glassresistor** [comment]
@bengolder thats good to know.  we can watch for that after the change.  we can also quickly restore an error.  I've tested the scripts on the production and staging buckets using the sync keys made for them.  Stages keys can't write to productions s3 bucket.  So as long as we don't add  ORIGIN_MEDIA_BUCKET_FOR_SYNC to production and the run download_data, which would over write the production database with whatever sql is in the sync bucket, we should be safe.  If Production doesn't have read access on the staging s3 bucket so it shouldn't be able to sync in the reverse direction.

Given all of those protections its probably still a good idea to make a database snapshot and mark the time(to revert s3 to) immediately before we run the commands on production and staging.  

We can use this command to see the last modified object in s3 and compare it before and after on production

`aws s3 ls $bucket_name --recursive | sort`

File: helper_commands/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (rossettistone) [inline helper_commands/admin.py]: deleteme

File: helper_commands/management/commands/download_data.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            's3', 'sync',
+            's3://%s' % settings.ORIGIN_MEDIA_BUCKET_FOR_SYNC,
+            's3://%s' % settings.AWS_STORAGE_BUCKET_NAME,
+        ]
+        aws_open(sync_s3)
+        download_s3 = [
+            settings.AWS_CLI_LOCATION,
+            's3', 'mv',
+            's3://%s/%s' % (
+                settings.SYNC_BUCKET,
```
Comment (rossettistone) [inline helper_commands/management/commands/download_data.py]: add comments to denote source and target

File: helper_commands/management/commands/upload_data.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class Command(BaseCommand):
+    help = str(
+        "[THIS IS NOT FOR USE ON PERSONAL MACHINES]"
+        "Dumps database and uploads to s3")
+
+    def handle(self, *args, **kwargs):
+        with open(settings.SYNC_FIXTURE_LOCATION, 'w+') as f:
+            management.call_command('dumpdata', stdout=f)
+        upload_command = [
+            settings.AWS_CLI_LOCATION,
```
Comment (rossettistone) [inline helper_commands/management/commands/upload_data.py]: add comments denoting source and target

**Pull request merged**